python/cudf/cudf/core/frame.py

```diff
@@ -894,6 +894,29 @@ def drop_duplicates(self, subset=None, keep="first", nulls_are_equal=True):
         result._copy_categories(self)
         return result
 
+    def replace(self, to_replace, replacement, inplace=False):
+        result_cols = []
+        for col in self._data.items():
+            if to_replace is None and replacement is None:
+                result_cols.append(col[1])
+            else:
+                try:
+                    (
+                        col_all_nan,
+                        col_replacement,
+                        col_to_replace,
+                    ) = _get_replacement_values(replacement, col, to_replace)
+
+                    result_cols.append(
+                        col[1].find_and_replace(
+                            col_to_replace, col_replacement, col_all_nan
+                        )
+                    )
+                except KeyError:
+                    result_cols.append(col[1])
+
+        return result_cols
```

Comment:
Any reason this can't return an instance of `type(self)` instead of a list of columns? That way more of the core logic is pushed down into `Frame`.